src/inim.nim

```diff
@@ -229,108 +235,102 @@ proc getPromptSymbol(): string =
+        indentLevel += 1
+        previouslyIndented = true
+
+    # Don't run yet if still on indent
+    if indentLevel != 0:
+        # Skip indent for first line
+        let n = if currentExpression.hasIndentTrigger(): 1 else: 0
+        tempIndentCode &= IndentSpaces.repeat(indentLevel-n) &
+            currentExpression & "\n"
+        return
 
-        # Compile buffer
-        let (output, status) = compileCode()
+    # Compile buffer
+    let (output, status) = compileCode()
 
-        # Succesful compilation, expression is valid
-        if status == 0:
-            compilationSuccess(currentExpression, output)
-        # Maybe trying to echo value?
-        elif "has to be discarded" in output and indentLevel == 0: #
-            bufferRestoreValidCode()
+    # Succesful compilation, expression is valid
+    if status == 0:
+        compilationSuccess(currentExpression, output)
+    # Maybe trying to echo value?
+    elif "has to be discarded" in output and indentLevel == 0: #
+        bufferRestoreValidCode()
 
-            # Save the current expression as an echo
-            currentExpression = "echo $" & currentExpression
-            buffer.writeLine(currentExpression)
-            buffer.flushFile()
-
-            let (echo_output, echo_status) = compileCode()
-            if echo_status == 0:
-                compilationSuccess(currentExpression, echo_output)
-            else:
-                # Show any errors in echoing the statement
-                indentLevel = 0
-                showError(echo_output)
-            # Roll back to not include the temporary echo line
-            bufferRestoreValidCode()
-        # Compilation error
+        # Save the current expression as an echo
+        currentExpression = fmt"echo $({currentExpression})"
+        buffer.writeLine(currentExpression)
+        buffer.flushFile()
+
+        let (echo_output, echo_status) = compileCode()
+        if echo_status == 0:
+            compilationSuccess(currentExpression, echo_output)
         else:
-            # Write back valid code to buffer
-            bufferRestoreValidCode()
+            # Show any errors in echoing the statement
             indentLevel = 0
-            showError(output)
+            showError(echo_output)
+        # Roll back to not include the temporary echo line
+        bufferRestoreValidCode()
+    # Compilation error
+    else:
+        # Write back valid code to buffer
+        bufferRestoreValidCode()
+        indentLevel = 0
+        showError(output)
 
-        # Clean up
-        tempIndentCode = ""
+    # Clean up
+    tempIndentCode = ""
 
-proc initApp*() =
+proc initApp*(nim, srcFile: string, showHeader: bool, flags = "") =
     ## Initialize the ``app` variable.
-    app.new()
-    app.nim = "nim"
-    app.srcFile = ""
-    app.showHeader = true
+    app = App(
+        nim: nim,
+        srcFile: srcFile,
+        showHeader: showHeader,
+        flags: flags
+    )
```

Comment:
@Yardanico This refactor broke the tests. I've opened up https://github.com/inim-repl/INim/issues/64, would you have some time to fix the test?